controller/search.go

```diff
@@ -129,3 +132,163 @@ func (c *SearchController) Spaces(ctx *app.SpacesSearchContext) error {
 		return ctx.OK(&response)
 	})
 }
+
+// Users runs the user search action.
+func (c *SearchController) Users(ctx *app.UsersSearchContext) error {
+
+	q := ctx.Q
+	if q == "" {
+		return ctx.BadRequest(goa.ErrBadRequest(fmt.Errorf("Empty search query not allowed")))
+	} else if q == "*" {
+		q = "" // Allow empty query if * specified
```

Comment:
this if statement is weird. I'd understand that a query e.g `/users` will return all the users, and a query `/users?q=XXXX` will return the users that match this pattern